astropy/io/ascii/core.py

```diff
@@ -835,6 +1005,18 @@ def get_data_lines(self, lines):
         else:  # Don't copy entire data lines unless necessary
             self.data_lines = data_lines
 
+    def get_data_lines_iter(self, source):
+        """
+        READ: return an iterator over source data lines.
+        """
+        start_line = _get_line_index(self.start_line, get_lines_iter(source))
```

Comment:
This feels inefficient, iterating over the whole list of lines three times.  Not quite sure how to avoid it (and maybe fine for a second go-over).